### PR TITLE
[apiRefactor] Fixing variable/method naming

### LINKED ISSUE #1914: Remove underscores from private/protected vars
Un-idiomatic in Typescript, since the `private` and `protected` keywords already indicate not to mess with them.

Make sure to retain underscores on variables shadowing a getter/setter.

### PR DESCRIPTION
**Pull request by rcchen**
Part 1 of indeterminate.

Fixes #1914

### CODE REVIEW

**Comment by aicioara** [comment]
Are you just ordering things alphabetically here? Can you give me a summary of what should be CR-ed?

**Comment by rcchen** [comment]
It's a mix of ordering things, removing underscores from things, and renaming things to better names. I should really be pushing more stuff before doing this pull request, I think you can touch the Abstract stuff without merge conflicts...

File: src/animators/baseAnimator.ts [hunk 1]
```diff
@@ -37,8 +36,8 @@ export module Animators {
      */
     public static DEFAULT_EASING = "exp-out";
 
-    private _duration: number;
     private _delay: number;
+    private _duration: number;
```
Comment (aicioara) [inline src/animators/baseAnimator.ts]: why do these still have _? It is confusing to read. I believe you should fix these in this PR as well

File: src/components/label.ts [hunk 2]
```diff
@@ -41,10 +41,10 @@ export module Components {
      * "right"]`. Defaults to `"center"`.
      * @returns {Label} The calling Label.
      */
-    public xAlign(alignment: string): Label {
+    public xAlign(alignment: string) {
```
Comment (aicioara) [inline src/components/label.ts]: Probably not part of this PR? There are more issues regarding signatures. In particular, comments should not have types in them anymore

File: src/components/plots/abstractPlot.ts [hunk 3]
```diff
@@ -23,17 +23,17 @@ export module Plots {
   }
 
   export class AbstractPlot extends Components.AbstractComponent {
-    protected _dataChanged = false;
-    protected _key2PlotDatasetKey: D3.Map<PlotDatasetKey>;
-    protected _datasetKeysInOrder: string[];
+    protected _animate: boolean = false;
```
Comment (aicioara) [inline src/components/plots/abstractPlot.ts]: Why is there a WS left?

File: src/components/plots/barPlot.ts [hunk 4]
```diff
@@ -5,17 +5,20 @@ export module Plots {
...
+
+    protected _isVertical: boolean;
+
     private _baseline: D3.Selection;
     private _baselineValue: number;
     private _barAlignmentFactor = 0.5;
-    protected _isVertical: boolean;
     private _barLabelFormatter: Formatter = Formatters.identity();
     private _barLabelsEnabled = false;
+    private _defaultFillColor: string;
```
Comment (aicioara) [inline src/components/plots/barPlot.ts]: how about these underscores?

**Comment by aicioara** [comment]
I can partially touch my stuff indeed. Stopped half way through the CR, as it needs more stuff added to be consistent

**Comment by bluong** [comment]
Feel free to use the sidebranch model off of the sidebranch model for cleaner organization

File: src/animators/baseAnimator.ts (hunk 1 above)
Comment (rcchen) [inline src/animators/baseAnimator.ts]: Chatted with @bluong and we are keeping underscores if there is a method that shares the same name. For example, because the getter method delay() exists, we keep _delay as a variable

**Comment by rcchen** [comment]
abstractComponent has been completely refactored, about halfway through Interactions at this point. Rotating off this to go work on something else, the rest of Interactions should be done later tonight, as well as Dispatchers.

File: src/components/abstractComponent.ts [hunk 5]
```diff
@@ -2,99 +2,153 @@
...
-
-    private _parentElement: ComponentContainer;
-    private _xAlignProportion = 0; // What % along the free space do we want to position (0 = left, .5 = center, 1 = right)
-    private _yAlignProportion = 0;
-    protected _fixedHeightFlag = false;
-    protected _fixedWidthFlag = false;
-    protected _isSetup = false;
+
+    protected _content: D3.Selection;
     protected _isAnchored = false;
```
Comment (bluong) [inline src/components/abstractComponent.ts]: Remove underscore here?

File: src/components/abstractComponent.ts [hunk 6]
```diff
@@ -2,99 +2,153 @@
...
+    protected isSetup = false;
 
+    private _height: number;
     private _hitBox: D3.Selection;
-    private _interactionsToRegister: Interaction[] = [];
-    private _boxes: D3.Selection[] = [];
-    private _boxContainer: D3.Selection;
-    private _rootSVG: D3.Selection;
-    private _isTopLevelComponent = false;
+    private _removed = false;
```
Comment (bluong) [inline src/components/abstractComponent.ts]: Remove underscore here?

File: src/components/abstractComponent.ts [hunk 7]
```diff
@@ -2,99 +2,153 @@
...
+    private _usedLastLayout = false;
```
Comment (bluong) [inline src/components/abstractComponent.ts]: Remove underscore here?

File: src/components/abstractComponent.ts [hunk 8]
```diff
@@ -119,222 +173,204 @@ module Plottable {
...
+    public parent(parentElement?: ComponentContainer): any {
+      if (parentElement === undefined) {
+        return this.parentElement;
+      }
+
+      this.detach();
+      this.parentElement = parentElement;
+    }
+
+    public requestedSpace(availableWidth: number, availableHeight: number): _SpaceRequest {
```
Comment (bluong) [inline src/components/abstractComponent.ts]: _SpaceRequest should be underscored

File: src/components/abstractComponent.ts [hunk 9]
```diff
@@ -347,263 +383,244 @@ module Plottable {
...
 
-    /**
-     * Checks if the Component has a fixed height or false if it grows to fill available space.
-     * Returns false by default on the base Component class.
-     *
-     * @returns {boolean} Whether the component has a fixed height.
-     */
-    public _isFixedHeight(): boolean {
-      return this._fixedHeightFlag;
+    public _requestedSpace(availableWidth: number, availableHeight: number): _SpaceRequest {
```
Comment (bluong) [inline src/components/abstractComponent.ts]: Should not be underscored and _SpaceRequest should not be underscored

File: src/components/axes/abstractAxis.ts [hunk 10]
```diff
@@ -187,6 +364,8 @@ module Plottable {
       return baselineAttrHash;
     }
 
+
+
```
Comment (bluong) [inline src/components/axes/abstractAxis.ts]: All the spaces...

File: src/components/axes/abstractAxis.ts [hunk 11]
```diff
@@ -224,112 +403,21 @@ module Plottable {
         case "right":
           tickMarkAttrHash["x2"] = tickLength;
           break;
-      }
+        }
```
Comment (bluong) [inline src/components/axes/abstractAxis.ts]: Unneeded spaces

File: src/components/axes/abstractAxis.ts [hunk 12]
```diff
@@ -224,112 +403,21 @@ module Plottable {
...
-        return this._endTickLength;
-      } else {
-        if (length < 0) {
-          throw new Error("end tick length must be positive");
-        }
-        this._endTickLength = length;
-        this._invalidateLayout();
-        return this;
-      }
+    protected _isHorizontal() {
```
Comment (bluong) [inline src/components/axes/abstractAxis.ts]: Shouldn't be underscored

File: src/components/axes/abstractAxis.ts [hunk 13]
```diff
@@ -340,117 +428,35 @@ module Plottable {
...
-     *
-     * @param {boolean} show Whether or not to show the first and last
-     * labels.
-     * @returns {Axis} The calling Axis.
-     */
-    public showEndTickLabels(show: boolean): Axis;
-    public showEndTickLabels(show?: boolean): any {
-      if (show == null) {
-        return this._showEndTickLabels;
+    protected _setDefaultAlignment() {
```
Comment (bluong) [inline src/components/axes/abstractAxis.ts]: remove underscore

File: src/components/axes/categoryAxis.ts [hunk 14]
```diff
@@ -89,19 +108,34 @@ export module Axes {
...
      * Measures the size of the ticks while also writing them to the DOM.
      * @param {D3.Selection} ticks The tick elements to be written to.
      */
-    private _drawTicks(axisWidth: number, axisHeight: number, scale: Scales.Category, ticks: D3.Selection) {
+    private drawTicks(axisWidth: number, axisHeight: number, scale: Scales.Category, ticks: D3.Selection) {
       var self = this;
       var xAlign: {[s: string]: string};
       var yAlign: {[s: string]: string};
-      switch (this.tickLabelAngle()) {
+      switch (this._tickLabelAngle) {
```
Comment (bluong) [inline src/components/axes/categoryAxis.ts]: Let's keep to the original implementation with of tickLabelAngle()

File: src/components/axes/categoryAxis.ts [hunk 15]
```diff
@@ -121,11 +155,11 @@ export module Axes {
         var height = self._isHorizontal() ? axisHeight - self._maxLabelTickLength() - self.tickLabelPadding() : bandWidth;
         var writeOptions = {
           selection: d3.select(this),
-          xAlign: xAlign[self.orient()],
-          yAlign: yAlign[self.orient()],
-          textRotation: self.tickLabelAngle()
+          xAlign: xAlign[self.orientation()],
+          yAlign: yAlign[self.orientation()],
+          textRotation: self._tickLabelAngle
```
Comment (bluong) [inline src/components/axes/categoryAxis.ts]: Keep to the original implementation of self.tickLabelAngle() unless you have a strong opinion against

File: src/components/axes/numericAxis.ts [hunk 16]
```diff
@@ -3,14 +3,12 @@
...
-
-    private _tickLabelPositioning = "center";
-    // Whether or not first/last tick label will still be displayed even if
-    // the label is cut off.
-    private _showFirstTickLabel = false;
-    private _showLastTickLabel = false;
-    private _measurer: SVGTypewriter.Measurers.Measurer;
-    private _wrapper: SVGTypewriter.Wrappers.Wrapper;
+    private measurer: SVGTypewriter.Measurers.Measurer;
+    // Whether or not first/last tick label will still be displayed even if he label is cut off.
```
Comment (bluong) [inline src/components/axes/numericAxis.ts]: the label*

File: src/components/dragBoxLayer.ts [hunk 17]
```diff
@@ -250,12 +250,12 @@ export module Components {
...
+      this.setResizableClasses(canResize);
       return this;
     }
 
     // Sets resizable classes. Overridden by subclasses that only resize in one dimension.
-    protected _setResizableClasses(canResize: boolean) {
+    protected setResizableClasses(canResize: boolean) {
       this.classed("x-resizable", canResize);
       this.classed("y-resizable", canResize);
     }
```
Comment (bluong) [inline src/components/dragBoxLayer.ts]: dragBoxLayer has a bunch of variables to affect

File: src/components/interpolatedColorLegend.ts [hunk 18]
```diff
@@ -98,13 +98,13 @@ export module Components {
...
      * @returns {InterpolatedColorLegend} The calling InterpolatedColorLegend.
      */
-    public orient(newOrientation: string): InterpolatedColorLegend;
-    public orient(newOrientation?: string): any {
+    public orientation(newOrientation: string): InterpolatedColorLegend;
+    public orientation(newOrientation?: string): any {
       if (newOrientation == null) {
         return this._orientation;
       } else {
         this._orientation = InterpolatedColorLegend._ensureOrientation(newOrientation);
```
Comment (bluong) [inline src/components/interpolatedColorLegend.ts]: _ensureOrientation should remove the underscore

File: src/components/interpolatedColorLegend.ts [hunk 19]
```diff
@@ -166,8 +166,8 @@ export module Components {
       return this._orientation !== "horizontal";
     }
 
-    public _doRender() {
-      super._doRender();
+    public doRender() {
+      super.doRender();
 
       var domain = this._scale.domain();
```
Comment (bluong) [inline src/components/interpolatedColorLegend.ts]: InterpolatedColorLegend has a bunch of variables to affect

File: src/components/legend.ts [hunk 20]
```diff
@@ -44,17 +44,17 @@ export module Components {
       }
 
       this._scale = colorScale;
-      this._scale.broadcaster.registerListener(this, () => this._invalidateLayout());
+      this._scale.broadcaster.registerListener(this, () => this.invalidateLayout());
```
Comment (bluong) [inline src/components/legend.ts]: this._scale should remove underscore

Comment (jtlan) [inline src/components/legend.ts]: Can't because `Legend` has a `scale()` getter/setter.

File: src/components/interpolatedColorLegend.ts [hunk 21]
```diff
@@ -40,12 +40,12 @@ export module Components {
         throw new Error("InterpolatedColorLegend requires a interpolatedColorScale");
       }
       this._scale = interpolatedColorScale;
-      this._scale.broadcaster.registerListener(this, () => this._invalidateLayout());
+      this._scale.broadcaster.registerListener(this, () => this.invalidateLayout());
```
Comment (bluong) [inline src/components/interpolatedColorLegend.ts]: _scale should not be underscored

File: src/components/legend.ts (hunk 20 above)
Comment (bluong) [inline src/components/legend.ts]: Whoa, did not realize.

File: src/components/plots/abstractXYPlot.ts [hunk 22]
```diff
@@ -115,65 +137,55 @@ module Plottable {
...
-    protected _updateXDomainer() {
+    protected projectorsReady() {
+      return this.projections["x"] && this.projections["y"];
+    }
+
+    protected updateXDomainer() {
       if (this._xScale instanceof QuantitativeScale) {
         var scale = <QuantitativeScale<any>> this._xScale;
-        if (!scale._userSetDomainer) {
+        if (!scale.setByUser) {
```
Comment (bluong) [inline src/components/plots/abstractXYPlot.ts]: Rename this back to userSetDomainer.  We will have a relook at this later.  Let's for now keep this scope as minimal as possible

File: src/components/plots/areaPlot.ts [hunk 23]
```diff
@@ -25,28 +24,48 @@ export module Plots {
...
     }
 
-    protected _getDrawer(key: string) {
+    protected getDrawer(key: string) {
       return new Plottable.Drawers.Area(key);
     }
 
-    protected _updateYDomainer() {
-      super._updateYDomainer();
+    protected _getResetYFunction() {
```
Comment (bluong) [inline src/components/plots/areaPlot.ts]: Remove underscore

File: src/components/plots/barPlot.ts [hunk 24]
```diff
@@ -3,19 +3,21 @@
...
-    protected static _BarAlignmentToFactor: {[alignment: string]: number} = {"left": 0, "center": 0.5, "right": 1};
-    protected static _DEFAULT_WIDTH = 10;
-    private static _BAR_WIDTH_RATIO = 0.95;
-    private static _SINGLE_BAR_DIMENSION_RATIO = 0.4;
-    private _baseline: D3.Selection;
-    private _baselineValue: number;
-    private _barAlignmentFactor = 0.5;
+    protected static BarAlignmentToFactor: {[alignment: string]: number} = {"left": 0, "center": 0.5, "right": 1};
+    protected static DEFAULT_WIDTH = 10;
     protected _isVertical: boolean;
```
Comment (bluong) [inline src/components/plots/barPlot.ts]: Remove underscore

File: src/components/plots/stackedAreaPlot.ts [hunk 25]
```diff
@@ -3,10 +3,9 @@
 module Plottable {
 export module Plots {
   export class StackedArea<X> extends Area<X> {
-
     private _isVertical: boolean;
```
Comment (bluong) [inline src/components/plots/stackedAreaPlot.ts]: Remove underscore

File: src/scales/quantitativeScale.ts [hunk 26]
```diff
@@ -2,15 +2,16 @@
 
 module Plottable {
   export class QuantitativeScale<D> extends Scale<D, number> {
-    protected _d3Scale: D3.Scale.QuantitativeScale;
-    private _numTicks = 10;
-    private _PADDING_FOR_IDENTICAL_DOMAIN = 1;
-    public _userSetDomainer: boolean = false;
+    public setByUser: boolean = false;
```
Comment (bluong) [inline src/scales/quantitativeScale.ts]: Let's hold back on renames

**Comment by bluong** [comment]
This pull request is quite... large.  It's encapsulating many ideas which make it very hard to review.  The most controversial thing from what I can gather is if we want this idea of methods being in exposure order + alphabetical order.

File: src/components/axes/categoryAxis.ts (hunk 14 above)
Comment (rcchen) [inline src/components/axes/categoryAxis.ts]: What's the difference? This one takes advantage of the private variable that can be accessed in the class; no reason to use the public getter method.

File: src/components/axes/categoryAxis.ts (hunk 15 above)
Comment (rcchen) [inline src/components/axes/categoryAxis.ts]: See above note.

**Comment by rcchen** [comment]
Closing this because it was a disaster, using a separate branch off API workflow now

File: src/components/abstractComponent.ts (hunk 8 above)
Comment (jtlan) [inline src/components/abstractComponent.ts]: ... but it is? Do you mean `requestedSpace()`?

**Comment by aicioara** [comment]
Might want to delete the branch as well?
